lib/simplifyExpression/functionsSearch/nthRoot.js

```diff
@@ -463,11 +479,10 @@ function isMultiplicationOfEqualNodes(node) {
     return false;
   }
 
-  // return if they are all equal nodes
-  return node.args.reduce((a, b) => {
-    return a.equals(b);
-  });
+  const terms = node.args.map(print);
 
+  // return if they are all equal nodes
+  return [...new Set(terms)].length === 1;
```

Comment:
I like it, I've used len(set(x))==1 to check for equality before. @aliang8 I'm wondering why this change was made though? Did the old way not work for a particular case?